packages/mui-material/src/Skeleton/Skeleton.js

```diff
@@ -166,6 +173,8 @@ const Skeleton = React.forwardRef(function Skeleton(inProps, ref) {
     ...props,
     animation,
     component,
+    size,
+    shape,
```

Comment:
Let's move [this](https://github.com/mui/material-ui/pull/38483/files#diff-772a1a6c6b82fcb6b5bee1c620c5e856d7eb3bc09ded68930028cee8ad55be31R80-R81) here:

```diff
+const size = size ?? variant === "text" ? variant : undefined
+const shape = shape ?? ["circular", "rectangular", "rounded"].includes(variant) ? variant : undefined

 const ownerState = {
   // ...
   size,
   shape,
-  variant,
   // ...
 }
```

That way we can remove the variant from the ownerState, removing it also from [here](https://github.com/mui/material-ui/pull/38483/files#diff-772a1a6c6b82fcb6b5bee1c620c5e856d7eb3bc09ded68930028cee8ad55be31R80-R81), [here](https://github.com/mui/material-ui/pull/38483/files#diff-772a1a6c6b82fcb6b5bee1c620c5e856d7eb3bc09ded68930028cee8ad55be31R20) and [here](https://github.com/mui/material-ui/pull/38483/files#diff-772a1a6c6b82fcb6b5bee1c620c5e856d7eb3bc09ded68930028cee8ad55be31R69).